### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by vol4onok**

### CODE REVIEW

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * @group LogEventActionTest
+ */
+class LogEventActionTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testExecuteWithLog(): void
+    {
+        $logEventAction = new LogEventAction($this->createEventLoggerMock($this->once()));
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:32]: Add
```php
// Arrange
// Act
// Assert 
``` 
sections

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+        $this->assertSame($context->getExitCode(), $result->getExitCode());
+    }
+
+    /**
+     * @return void
+     */
+    public function testExecuteWithoutLog(): void
+    {
+        $logEventAction = new LogEventAction($this->createEventLoggerMock($this->never()));
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:44]: The same as here https://github.com/spryker-sdk/sdk/pull/61/files#r786651379

File: tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+ * @group CommandExecutorTest
+ */
+class CommandExecutorTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testExecute(): void
+    {
+        $context = $this->createContextMock();
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php:32]: The same as here https://github.com/spryker-sdk/sdk/pull/61/files#r786651379
Please look through your tests and add comments

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        $context = $this->createContextMock();
+        $result = $logEventAction->execute($this->createCommandMock(), $context, '');
+
+        $this->assertSame($context->getExitCode(), $result->getExitCode());
+    }
+
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Entity\ContextInterface
+     */
+    protected function createContextMock(): mixed
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:54]: Please, specify the return type

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        $context->method('getExitCode')
+            ->willReturn(0);
+
+        return $context;
+    }
+
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Entity\TaskInterface
+     */
+    protected function createTaskMock(): mixed
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:69]: Please, specify the return type

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        return $this->createMock(TaskInterface::class);
+    }
+
+    /**
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Entity\CommandInterface
+     */
+    protected function createCommandMock(): mixed
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:77]: Please, specify the return type

File: tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    {
+        return $this->createMock(CommandInterface::class);
+    }
+
+    /**
+     * @param \PHPUnit\Framework\MockObject\Rule\InvocationOrder $invocationRule
+     *
+     * @return \PHPUnit\Framework\MockObject\MockObject|\SprykerSdk\SdkContracts\Logger\EventLoggerInterface
+     */
+    protected function createEventLoggerMock(InvocationOrder $invocationRule): mixed
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/AfterCommandExecutedAction/LogEventActionTest.php:87]: Please, look through tests and fix return types for methods

File: tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+        if ($invocationRule === null) {
+            $invocationRule = $this->once();
+        }
+        $commandRunner = $this->createMock(CommandRunnerInterface::class);
+        $commandRunner
+            ->expects($invocationRule)
+            ->method('canHandle')
+            ->willReturn($canHandle);
+        $commandRunner
+            ->expects(!$canHandle || $invocationRule->isNever() || $context->isDryRun() ? $this->never() : $this->once())
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php:94]: Let's keep tests simple, without complicated conditions, etc
Please rewrite it

File: tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    protected function createContextMock(bool $isDry = false): ContextInterface
+    {
+        $context = $this->createMock(ContextInterface::class);
+        $context->method('isDryRun')
+            ->willReturn($isDry);
+
+        $context->method('getExitCode')
+            ->willReturn(0);
+
+        $context->expects($isDry ? $this->atLeast(2) : $this->never())
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php:115]: Please, avoid magic numbers

File: tests/Sdk/Tests/Core/Application/Service/Violation/ViolationConverterResolverTest.php [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+ * @group ViolationConverterResolverTest
+ */
+class ViolationConverterResolverTest extends Unit
+{
+    /**
+     * @return void
+     */
+    public function testResolve(): void
+    {
+        $violationConverterResolver = new ViolationConverterResolver($this->createConverterRegistryMock());
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/Violation/ViolationConverterResolverTest.php:32]: You can use `setUp` method to create `ViolationConverterResolver`

**Review by Kyrylo-Meretskyi — CHANGES_REQUESTED** [review]

File: tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php (hunk 9 above)
Comment (vol4onok) [reply tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php:115]: it's expected numbers

File: tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php (hunk 8 above)
Comment (vol4onok) [reply tests/Sdk/Tests/Core/Application/Service/CommandExecutorTest.php:94]: preconditions for expectation. it's unit test

File: tests/Sdk/Tests/Core/Application/Service/Violation/ViolationReportMergerTest.php [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
+    /**
+     * @param string $prefix
+     *
+     * @return \SprykerSdk\SdkContracts\Violation\PackageViolationReportInterface
+     */
+    protected function createPackageViolationReport(string $prefix = ''): PackageViolationReportInterface
+    {
+        return new PackageViolationReport(
+            'package' . $prefix,
```
Comment (Kyrylo-Meretskyi) [inline tests/Sdk/Tests/Core/Application/Service/Violation/ViolationReportMergerTest.php:75]: Please, rename `$prefix` to `$postfix` or use the variable properly according to naming

**Review by Kyrylo-Meretskyi — CHANGES_REQUESTED** [review]

**Review by Kyrylo-Meretskyi — APPROVED** [review]

**Pull request merged**